test/Microsoft.AspNet.Mvc.DataAnnotations.Test/DataAnnotationsModelValidatorProviderTest.cs

```diff
@@ -39,6 +39,27 @@ public void GetValidators_ReturnsValidatorForIValidatableObject()
 #endif
 
         [Fact]
+        public void GetValidators_InsertsRequiredValidatorsFirst()
+        {
+            var provider = new DataAnnotationsModelValidatorProvider(
+                new TestOptionsManager<MvcDataAnnotationsLocalizationOptions>(),
+                stringLocalizerFactory: null);
+            var metadata = _metadataProvider.GetMetadataForProperty(
+                typeof(ClassWithProperty),
+                "PropertyWithMultipleValidationAttributes");
+
+            var providerContext = new ModelValidatorProviderContext(metadata);
+
+            // Act
+            provider.GetValidators(providerContext);
+
+            // Assert
+            Assert.Equal(4, providerContext.Validators.Count);
+            Assert.IsAssignableFrom<RequiredAttribute>((providerContext.Validators[0] as DataAnnotationsModelValidator).Attribute);
+            Assert.IsAssignableFrom<RequiredAttribute>((providerContext.Validators[1] as DataAnnotationsModelValidator).Attribute);
```

Comment:
Get used to using a c-style cast in these kind of cases. `((DataAnnotationsModelValidator)providerContext.Validators[1]).Attribute` instead of  `(providerContext.Validators[1] as DataAnnotationsModelValidator).Attribute`

It's more semantically correct, and fail at the case instead with a mysterious nullref later.